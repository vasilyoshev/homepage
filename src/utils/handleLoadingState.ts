import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
