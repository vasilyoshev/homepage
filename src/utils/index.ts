export * from './handleLoadingState';
export * from './testUtils';
