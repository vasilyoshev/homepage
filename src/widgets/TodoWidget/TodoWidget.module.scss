.todoWidget {
  background-color: #dde0e0;
  border: 1px solid rgb(88, 85, 85);
}
