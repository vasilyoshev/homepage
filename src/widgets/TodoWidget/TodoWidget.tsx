import React, { useState } from 'react';
import { AddTodoForm, TodoList } from 'components';
import { Todo } from 'interfaces';
import { Card, CardContent } from '@material-ui/core';
import styles from './TodoWidget.module.scss';

export const TodoWidget: React.FC = () => {
  const initialTodos: Array<Todo> = [
    {
      text: 'Walk the dog',
      complete: true,
    },
    {
      text: 'Make app',
      complete: false,
    },
  ];
  const [todos, setTodos] = useState<Array<Todo>>(initialTodos);
  const toggleComplete = (selectedTodo: Todo): void => {
    const updatedTodos = todos.map((todo) => {
      if (todo === selectedTodo) {
        return { ...todo, complete: !todo.complete };
      }
      return todo;
    });
    setTodos(updatedTodos);
  };
  const addTodo = (newTodo: string): void => {
    newTodo.trim() !== '' &&
      setTodos([...todos, { text: newTodo, complete: false }]);
  };
  const removeTodo = (text: string): void => {
    setTodos(todos.filter((t) => t.text !== text));
  };
  return (
    <Card className={styles.todoWidget}>
      <CardContent>
        <TodoList todos={todos} toggleComplete={toggleComplete} removeTodo={removeTodo} />
        <AddTodoForm addTodo={addTodo} />
      </CardContent>
    </Card>
  );
};
