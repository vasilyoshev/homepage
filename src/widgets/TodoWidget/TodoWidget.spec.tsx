import { render } from 'utils';
import { TodoWidget } from './TodoWidget';

describe('TodoWidget', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<TodoWidget />);
    expect(asFragment()).toMatchSnapshot();
  });
});
