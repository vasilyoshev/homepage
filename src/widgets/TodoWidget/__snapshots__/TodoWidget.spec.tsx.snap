// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoWidget renders successfully 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root todoWidget MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root complete MuiInputLabel-animated"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorPrimary MuiIconButton-edgeStart"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-4"
              data-indeterminate="false"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
        </span>
        <p
          class="MuiTypography-root todoText MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-displayInline"
        >
          Walk the dog
        </p>
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root"
          data-testid="removeTodoItem"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            onClick=
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </label>
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary MuiIconButton-edgeStart"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-4"
              data-indeterminate="false"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </span>
        <p
          class="MuiTypography-root todoText MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-displayInline"
        >
          Make app
        </p>
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root"
          data-testid="removeTodoItem"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            onClick=
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </label>
      <div
        class="MuiFormGroup-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
          >
            add new task
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-7"
              >
                <span>
                  add new task
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Add Todo
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
